### ISSUE TITLE
[BMW Connected Drive] Authorization chain from BMW changed

### ISSUE THREAD

**Issue by weymann** [openhab/openhab-addons#11213]
Several [posts in the Community](https://community.openhab.org/t/bmw-connecteddrive-binding/105124/118) shows that the current released **BMW Connected Drive binding** doesn't work anymore. BMW exchanged in the meanwhile their **Connected Drive** Android App with the **MyBMW** App and also the Authorization workflow changed.

**Comment by pboettch** [comment]
Are you sure this is in authentication problem and not "just" a new data-model which is being used?

I have small scripts which I used before switching to OpenHAB and it still can login, but is not able to fetch the data anylonger.

**Comment by McMornan** [comment]
Hi,

the author of this module seems to have solved the issue: https://github.com/hdurdle/MMM-BMWConnected

I use it for my magic mirror and it is working fine.

Lars

**Comment by pboettch** [comment]
With my super small script I can still login to get the auth-token, but then I cannot fetch any data anymore.

It also seems that the web-version of the remote-cockpit has been removed from the connected-drive website.

**Comment by weymann** [comment]
@pboettch Yes, we're sure authorization fails. This is shown in the logs provided by other users and I observed the same in my installation.
```
2021-08-05 16:40:52.497 [DEBUG] [internal.handler.ConnectedDriveProxy] - Authorization status 400 reason Bad Request
2021-08-05 16:40:52.639 [DEBUG] [internal.handler.ConnectedDriveProxy] - HTTP Error https://b2vapi.bmwgroup.com/webapi/v1/user/vehicles - 400 - Bad Requestnull
```

@McMornan There several projects in github handling BMW ConnectedDrive. The one you mentioned will perhaps work for **Rest of Worl** but I'm pretty sure for US and China it will fail because they need different auth servers. In the binding README.md I refer to [bimmer_connected](https://github.com/bimmerconnected/bimmer_connected) because it seems it's the most active repo. In addition they are already preparing the new **MyBMW** app adaptions which  replaced the **Connected Drive** smartphone app. I'm pretty sure there will be additional changes in future.

I fixed this issue already and provided the [prototype in the community](https://community.openhab.org/t/bmw-connecteddrive-binding/105124). Community users already confirmed it's working. I still have some difficulties with Jetty HttpClient - that's why I didn't raise a pull request till now.
